analytics/runner.go

```diff
@@ -1,14 +1,15 @@
 package analytics
 
 import (
+	"github.com/prebid/prebid-server/v2/config"
 	"github.com/prebid/prebid-server/v2/privacy"
 )
 
 type Runner interface {
-	LogAuctionObject(*AuctionObject, privacy.ActivityControl)
-	LogVideoObject(*VideoObject, privacy.ActivityControl)
+	LogAuctionObject(*AuctionObject, privacy.ActivityControl, config.AccountPrivacy)
+	LogVideoObject(*VideoObject, privacy.ActivityControl, config.AccountPrivacy)
 	LogCookieSyncObject(*CookieSyncObject)
 	LogSetUIDObject(*SetUIDObject)
-	LogAmpObject(*AmpObject, privacy.ActivityControl)
+	LogAmpObject(*AmpObject, privacy.ActivityControl, config.AccountPrivacy)
```

Comment:
I think we should consider an alternative that avoids passing `config.AccountPrivacy` as a parameter. Instead, it seems like we could update the `ActivityControl` object by adding a new field that stores the applicable config ( `IPv4Config` and `IPv6Config`) on it. We could then update the factory function `NewActivityControl` to set the new field similar to how plans are pulled from the `AccountPrivacy` object and stored on the `ActivityControl` object.
In this case, any activity control infrastructure configuration pulled from the account is stored on the object.

Reply:
Refactored. Looks so much better now!